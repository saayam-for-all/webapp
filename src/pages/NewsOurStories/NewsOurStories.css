/* NewsOurStories Page Styles */

.news-our-stories-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  min-height: 100vh;
  width: 100%;
  background: #fff;
}

.news-our-stories-hero {
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 2.5rem 0;
  background: #fff;
}

.news-our-stories-title {
  font-size: 2rem;
  font-weight: bold;
  margin-bottom: 1rem;
  text-align: center;
}

@media (min-width: 768px) {
  .news-our-stories-title {
    font-size: 2.5rem;
  }
}

.news-our-stories-subtitle {
  color: #374151;
  font-size: 1rem;
  max-width: 56rem;
  width: 100%;
  text-align: center;
  margin-bottom: 0.5rem;
}

.news-our-stories-desc {
  color: #6b7280;
  font-size: 0.95rem;
  max-width: 56rem;
  width: 100%;
  text-align: center;
}

.news-our-stories-section {
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 3rem;
  padding-bottom: 2.5rem;
}

.news-our-stories-story {
  width: 100%;
  max-width: 56rem;
}

.news-our-stories-img-lg {
  width: 100%;
  height: 400px;
  object-fit: cover;
  margin-bottom: 1rem;
  border-radius: 10px;
}

.news-our-stories-img-sm {
  width: 100%;
  height: 300px;
  object-fit: cover;
  margin-bottom: 1rem;
  border-radius: 10px;
}

.news-our-stories-date {
  font-size: 0.75rem;
  color: #6b7280;
  margin-bottom: 0.25rem;
}

.news-our-stories-story-title {
  font-size: 1.25rem;
  font-weight: 600;
  margin-bottom: 0.5rem;
}

@media (min-width: 768px) {
  .news-our-stories-story-title {
    font-size: 1.5rem;
  }
}

.news-our-stories-story-desc {
  color: #374151;
  margin-bottom: 0.5rem;
}

.news-our-stories-link {
  color: #2563eb;
  font-size: 0.95rem;
  font-weight: 500;
  text-decoration: underline;
  cursor: pointer;
}

.news-our-stories-row {
  display: flex;
  flex-direction: column;
  gap: 2rem;
  width: 100%;
  max-width: 56rem;
}

@media (min-width: 768px) {
  .news-our-stories-row {
    flex-direction: row;
  }
}

.news-our-stories-col {
  flex: 1;
}

.news-our-stories-cta {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: 4rem 0;
  background: #fff;
  text-align: center;
  width: 100%;
  margin-top: 1rem;
}

.news-our-stories-cta-title {
  font-size: 1.5rem;
  font-weight: bold;
  margin-bottom: 0.25rem;
}

.news-our-stories-cta-desc {
  color: #9ca3af;
  font-size: 1rem;
  max-width: 28rem;
  margin-bottom: 1rem;
}

.news-our-stories-cta-btn {
  background-color: #2a72ff;
  color: #fff;
  font-size: 1rem;
  font-weight: bold;
  padding: 0.75rem 2rem;
  border-radius: 9999px;
  border: none;
  cursor: pointer;
  transition: background 0.2s;
}

.news-our-stories-cta-btn:hover {
  background-color: #1e5dcc;
}
