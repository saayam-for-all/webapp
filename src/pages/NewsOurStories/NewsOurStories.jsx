import React from "react";
import seventeenMileWalk from "../../assets/news_our_stoires/17_Mile_walk.png";
import communityUpliftment from "../../assets/news_our_stoires/community_upliftment.png";
import tieCon from "../../assets/news_our_stoires/tie_con.png";
import { useNavigate } from "react-router-dom";
import "./NewsOurStories.css";
import { useTranslation } from "react-i18next";

const stories = [
  {
    date: "05/02/2025",
    title: "Saayam for All CEO Represents at TiEcon AiVerse 2025, Santa Clara",
    image: tieCon,
    description:
      "Our CEO, Rao K Bhehanabobla recently attended TiEcon 2025 at the Santa Clara Convention Center, one of the premier global gatherings for entrepreneurs and innovators. He shared Saayam's journey in leveraging technology to build a support-driven community platform. The event spotlighted Saayam's mission and growing influence in the social impact ecosystem.",
    link: "#",
  },
  {
    date: "05/02/2025",
    title:
      "Saayam x Sri Madhusudan Sai: Collaboration for Community Upliftment",
    image: communityUpliftment,
    description:
      "Saayam is proud to collaborate with Sri Madhusudan Sai, a global humanitarian known for his work in healthcare, education, and rural upliftment. This partnership strengthens our shared commitment to serving communities with compassion, integrity, and purpose.",
    link: "#",
  },
  {
    date: "05/02/2025",
    title: "In Step with the Community: A 17-Mile Walk in San Ramon",
    image: seventeenMileWalk,
    description:
      "Our CEO and Satesth Mudrala recently participated in a 17-mile walk through San Ramon, California, championing wellness, unity, and public service. Their commitment to engaging with the local community exemplifies Saayam's mission of stepping forward—both literally and figuratively—for meaningful impact.",
    link: "#",
  },
];

export default function NewsOurStories() {
  const { t } = useTranslation();
  const navigate = useNavigate();
  return (
    <div className="news-our-stories-container px-4 md:px-0">
      {/* Hero Section */}
      <section className="news-our-stories-hero">
        <h1 className="news-our-stories-title">{t("News: Our Stories")}</h1>
        <p className="news-our-stories-subtitle">
          {t(
            "Explore how Saayam for All is making headlines and gaining recognition for its work in uplifting communities, empowering volunteers, and building an inclusive support network.",
          )}
        </p>
        <p className="news-our-stories-desc">
          {t(
            "From local stories to national features, discover how our mission is resonating beyond the platform and into the world.",
          )}
        </p>
      </section>

      {/* Stories Section */}
      <section className="news-our-stories-section">
        {/* First story: full width image and text */}
        <div className="news-our-stories-story">
          <img
            src={stories[0].image}
            alt={stories[0].title}
            className="news-our-stories-img-lg"
          />
          <div className="news-our-stories-date">{stories[0].date}</div>
          <h2 className="news-our-stories-story-title">
            {t(stories[0].title)}
          </h2>
          <p className="news-our-stories-story-desc">
            {t(stories[0].description)}
          </p>
          <a href={stories[0].link} className="news-our-stories-link">
            {t("Read More")}
          </a>
        </div>
        {/* Next two stories: two columns on desktop, stacked on mobile, no card effect */}
        <div className="news-our-stories-row">
          {stories.slice(1).map((story) => (
            <div key={story.title} className="news-our-stories-col">
              <img
                src={story.image}
                alt={story.title}
                className="news-our-stories-img-sm"
              />
              <div className="news-our-stories-date">{story.date}</div>
              <h2 className="news-our-stories-story-title">{t(story.title)}</h2>
              <p className="news-our-stories-story-desc">
                {t(story.description)}
              </p>
              <a href={story.link} className="news-our-stories-link">
                {t("Read More")}
              </a>
            </div>
          ))}
        </div>
      </section>

      {/* Call to Action Section */}
      <section className="news-our-stories-cta">
        <h2 className="news-our-stories-cta-title">{t("Want to join us?")}</h2>
        <p className="news-our-stories-cta-desc">
          {t(
            "Chat with our community and get in touch with different charity organizations!",
          )}
        </p>
        <button
          className="news-our-stories-cta-btn"
          onClick={() => navigate("/contact")}
        >
          {t("Join the community")}
        </button>
      </section>
    </div>
  );
}
