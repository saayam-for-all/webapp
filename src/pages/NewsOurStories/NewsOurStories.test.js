import { render } from "@testing-library/react";
import NewsOurStories from "./NewsOurStories";

describe("NewsOurStories", () => {
  it("renders correctly", () => {
    const tree = render(<NewsOurStories />);
    expect(tree).toMatchSnapshot();
  });
});
