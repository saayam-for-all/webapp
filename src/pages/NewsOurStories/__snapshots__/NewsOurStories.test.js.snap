// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsOurStories renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="news-our-stories-container px-4 md:px-0"
      >
        <section
          class="news-our-stories-hero"
        >
          <h1
            class="news-our-stories-title"
          >
            mockTranslate(News: Our Stories)
          </h1>
          <p
            class="news-our-stories-subtitle"
          >
            mockTranslate(Explore how Saayam for All is making headlines and gaining recognition for its work in uplifting communities, empowering volunteers, and building an inclusive support network.)
          </p>
          <p
            class="news-our-stories-desc"
          >
            mockTranslate(From local stories to national features, discover how our mission is resonating beyond the platform and into the world.)
          </p>
        </section>
        <section
          class="news-our-stories-section"
        >
          <div
            class="news-our-stories-story"
          >
            <img
              alt="Saayam for All CEO Represents at TiEcon AiVerse 2025, Santa Clara"
              class="news-our-stories-img-lg"
              src=""
            />
            <div
              class="news-our-stories-date"
            >
              05/02/2025
            </div>
            <h2
              class="news-our-stories-story-title"
            >
              mockTranslate(Saayam for All CEO Represents at TiEcon AiVerse 2025, Santa Clara)
            </h2>
            <p
              class="news-our-stories-story-desc"
            >
              mockTranslate(Our CEO, Rao K Bhehanabobla recently attended TiEcon 2025 at the Santa Clara Convention Center, one of the premier global gatherings for entrepreneurs and innovators. He shared Saayam's journey in leveraging technology to build a support-driven community platform. The event spotlighted Saayam's mission and growing influence in the social impact ecosystem.)
            </p>
            <a
              class="news-our-stories-link"
              href="#"
            >
              mockTranslate(Read More)
            </a>
          </div>
          <div
            class="news-our-stories-row"
          >
            <div
              class="news-our-stories-col"
            >
              <img
                alt="Saayam x Sri Madhusudan Sai: Collaboration for Community Upliftment"
                class="news-our-stories-img-sm"
                src=""
              />
              <div
                class="news-our-stories-date"
              >
                05/02/2025
              </div>
              <h2
                class="news-our-stories-story-title"
              >
                mockTranslate(Saayam x Sri Madhusudan Sai: Collaboration for Community Upliftment)
              </h2>
              <p
                class="news-our-stories-story-desc"
              >
                mockTranslate(Saayam is proud to collaborate with Sri Madhusudan Sai, a global humanitarian known for his work in healthcare, education, and rural upliftment. This partnership strengthens our shared commitment to serving communities with compassion, integrity, and purpose.)
              </p>
              <a
                class="news-our-stories-link"
                href="#"
              >
                mockTranslate(Read More)
              </a>
            </div>
            <div
              class="news-our-stories-col"
            >
              <img
                alt="In Step with the Community: A 17-Mile Walk in San Ramon"
                class="news-our-stories-img-sm"
                src=""
              />
              <div
                class="news-our-stories-date"
              >
                05/02/2025
              </div>
              <h2
                class="news-our-stories-story-title"
              >
                mockTranslate(In Step with the Community: A 17-Mile Walk in San Ramon)
              </h2>
              <p
                class="news-our-stories-story-desc"
              >
                mockTranslate(Our CEO and Satesth Mudrala recently participated in a 17-mile walk through San Ramon, California, championing wellness, unity, and public service. Their commitment to engaging with the local community exemplifies Saayam's mission of stepping forward—both literally and figuratively—for meaningful impact.)
              </p>
              <a
                class="news-our-stories-link"
                href="#"
              >
                mockTranslate(Read More)
              </a>
            </div>
          </div>
        </section>
        <section
          class="news-our-stories-cta"
        >
          <h2
            class="news-our-stories-cta-title"
          >
            mockTranslate(Want to join us?)
          </h2>
          <p
            class="news-our-stories-cta-desc"
          >
            mockTranslate(Chat with our community and get in touch with different charity organizations!)
          </p>
          <button
            class="news-our-stories-cta-btn"
          >
            mockTranslate(Join the community)
          </button>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="news-our-stories-container px-4 md:px-0"
    >
      <section
        class="news-our-stories-hero"
      >
        <h1
          class="news-our-stories-title"
        >
          mockTranslate(News: Our Stories)
        </h1>
        <p
          class="news-our-stories-subtitle"
        >
          mockTranslate(Explore how Saayam for All is making headlines and gaining recognition for its work in uplifting communities, empowering volunteers, and building an inclusive support network.)
        </p>
        <p
          class="news-our-stories-desc"
        >
          mockTranslate(From local stories to national features, discover how our mission is resonating beyond the platform and into the world.)
        </p>
      </section>
      <section
        class="news-our-stories-section"
      >
        <div
          class="news-our-stories-story"
        >
          <img
            alt="Saayam for All CEO Represents at TiEcon AiVerse 2025, Santa Clara"
            class="news-our-stories-img-lg"
            src=""
          />
          <div
            class="news-our-stories-date"
          >
            05/02/2025
          </div>
          <h2
            class="news-our-stories-story-title"
          >
            mockTranslate(Saayam for All CEO Represents at TiEcon AiVerse 2025, Santa Clara)
          </h2>
          <p
            class="news-our-stories-story-desc"
          >
            mockTranslate(Our CEO, Rao K Bhehanabobla recently attended TiEcon 2025 at the Santa Clara Convention Center, one of the premier global gatherings for entrepreneurs and innovators. He shared Saayam's journey in leveraging technology to build a support-driven community platform. The event spotlighted Saayam's mission and growing influence in the social impact ecosystem.)
          </p>
          <a
            class="news-our-stories-link"
            href="#"
          >
            mockTranslate(Read More)
          </a>
        </div>
        <div
          class="news-our-stories-row"
        >
          <div
            class="news-our-stories-col"
          >
            <img
              alt="Saayam x Sri Madhusudan Sai: Collaboration for Community Upliftment"
              class="news-our-stories-img-sm"
              src=""
            />
            <div
              class="news-our-stories-date"
            >
              05/02/2025
            </div>
            <h2
              class="news-our-stories-story-title"
            >
              mockTranslate(Saayam x Sri Madhusudan Sai: Collaboration for Community Upliftment)
            </h2>
            <p
              class="news-our-stories-story-desc"
            >
              mockTranslate(Saayam is proud to collaborate with Sri Madhusudan Sai, a global humanitarian known for his work in healthcare, education, and rural upliftment. This partnership strengthens our shared commitment to serving communities with compassion, integrity, and purpose.)
            </p>
            <a
              class="news-our-stories-link"
              href="#"
            >
              mockTranslate(Read More)
            </a>
          </div>
          <div
            class="news-our-stories-col"
          >
            <img
              alt="In Step with the Community: A 17-Mile Walk in San Ramon"
              class="news-our-stories-img-sm"
              src=""
            />
            <div
              class="news-our-stories-date"
            >
              05/02/2025
            </div>
            <h2
              class="news-our-stories-story-title"
            >
              mockTranslate(In Step with the Community: A 17-Mile Walk in San Ramon)
            </h2>
            <p
              class="news-our-stories-story-desc"
            >
              mockTranslate(Our CEO and Satesth Mudrala recently participated in a 17-mile walk through San Ramon, California, championing wellness, unity, and public service. Their commitment to engaging with the local community exemplifies Saayam's mission of stepping forward—both literally and figuratively—for meaningful impact.)
            </p>
            <a
              class="news-our-stories-link"
              href="#"
            >
              mockTranslate(Read More)
            </a>
          </div>
        </div>
      </section>
      <section
        class="news-our-stories-cta"
      >
        <h2
          class="news-our-stories-cta-title"
        >
          mockTranslate(Want to join us?)
        </h2>
        <p
          class="news-our-stories-cta-desc"
        >
          mockTranslate(Chat with our community and get in touch with different charity organizations!)
        </p>
        <button
          class="news-our-stories-cta-btn"
        >
          mockTranslate(Join the community)
        </button>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
